File: app-vue/src/api/order.ts
import request from './request'

export interface Order {
  id: number
  orderNo: string
  userId: number
  goodsId: number
  activityId: number
  quantity: number
  price: number
  status: number
  queueId: string
  createTime: string
}

export const orderApi = {
  getMyOrders() {
    return request.get<Order[]>('/order/my')
  },
  payOrder(orderNo: string) {
    return request.post<Boolean>(`/order/${orderNo}/pay`)
  },
  cancelOrder(orderNo: string) {
    return request.post<Boolean>(`/order/${orderNo}/cancel`)
  }
}

File: springboot/src/main/java/cn/coderstory/springboot/order/entity/Order.java
package cn.coderstory.springboot.order.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("seckill_order")
public class Order {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String orderNo;
    private Long userId;
    private Long goodsId;
    private Long activityId;
    private Integer quantity;
    private BigDecimal price;
    private Integer status;
    private String queueId;
    private LocalDateTime paymentTime;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
    @TableLogic
    private Integer deleted;
}
